app/api/files/specs/exportRoutes.spec.ts

```diff
@@ -0,0 +1,105 @@
+import { Writable } from 'stream';
+import request from 'supertest';
+import { setUpApp } from 'api/utils/testingRoutes';
+import search from 'api/search/search';
+import db from 'api/utils/testing_db';
+import csvExporter from 'api/csv/csvExporter';
+import * as filesystem from 'api/files/filesystem';
+
+import routes from '../exportRoutes';
+import { NextFunction, Request, Response } from 'express';
+import { User } from '../../users/usersModel';
+
+jest.mock('api/csv/csvExporter');
+
+function assertDownloaded(res: any) {
+  expect(res.header['content-type']).toMatch('text/csv');
+  expect(res.header['content-disposition']).toMatch(/^attachment; filename=(.*)/);
+}
+
+function assertExport(mockCall: any, searchResults: any, types: any, options: any) {
+  expect(mockCall[0]).toEqual(searchResults);
+  expect(mockCall[1]).toEqual(types);
+  expect(mockCall[2] instanceof Writable).toBe(true);
+  expect(mockCall[3]).toEqual(options);
+}
+
+describe('export routes', () => {
+  describe('/api/export', () => {
+    let exportMock: any;
+
+    beforeAll(() => {
```

Comment:
This is a weird scenario, and works because you have a single `it` on this test.  If someone else includes a previous `it` for whatever reason, the `exportMock.mock.calls[0]` on line 99 will fail as there is a single mock shared by all tests.

That is not a good practice.  My suggestion is to move this into the 'before each' so that you get the jest spy reset every time you run a test and, therefore, adding further tests (or running them on different orders!) would not affect this spy.

Reply:
Makes total sense, good catch, thank you!